### PR TITLE
Feature/backend/schedule

### PR DESCRIPTION
**Pull request by kimchhengheng**
# Description

- create the employee schedule
- get all employees' schedule
- delete employee schedule
- update employee schedule

---

# Related Jira Issue (ID/URL)

https://rtterror.atlassian.net/jira/software/projects/RTTER/boards/1?selectedIssue=RTTER-92

# Related GitHub Issues

- This fixes #5.

# Checklist

- [x] Code compiles correctly
- [x] Code is covered by current tests or new tests
- [x] Code is formatted correctly
- [x] Extended the README / documentation, if necessary

---

# Comment/Discussion

_Add anything else._

### CODE REVIEW

File: backend/src/services/schedule.service.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        appointmentsList = appointmentsList.filter(
+            (item) =>{
+                let date= item.datetime.toISOString().slice(0,10);
+                console.log({date: date, start:startDate,end:endDate})
+                console.log(date>=startDate && date<=endDate)
+                return date>=startDate && date<=endDate;
+            }
+        );
+        return appointmentsList;
+    }
```
Comment (successsb89) [inline backend/src/services/schedule.service.ts:86]: what is the behind logic of calculating salary based on salary formula, parameters and pay period? how we are going to customize the compensation for each employee?

**Comment by successsb89** [comment]
it did not pass the all tests. please check it out.

File: backend/src/services/schedule.service.ts (hunk 1 above)
Comment (kimchhengheng) [reply backend/src/services/schedule.service.ts:86]: the calculation is done on the frontend side. The customization is vary by the method of calculation and the parameter from the User.  
image

Comment (successsb89) [reply backend/src/services/schedule.service.ts:86]: yes, I saw that from PR #66 Thanks. good work!

**Review by successsb89 — APPROVED** [review]
issues are resolved, approved.

**Review by yuxiang-zhang — APPROVED** [review]
code is clean and automated checks passed

**Pull request merged**